### PR TITLE
[READY]Addition of new lavaland ruin that contains The Pioneer Stone

### PR DESCRIPTION
**Pull request by Djiq**
## About The Pull Request
This PR adds only 2 new things. A new lavaland ruin , and a new vehicle : pioneer stone. Pioneer stone is a quite good locomotion device. Pioneer camp has a couple of references to spongebob, such as being filled with crabs.
## Why It's Good For The Game
I think that it would be best for the game to contain the pioneer stone. pioneers used them to ride for miles, so can miners now !
## Changelog
:cl:Edge
add: New lavaland ruin: Pioneer Camp
add: New vehicle : The Pioneer Stone
/:cl:

### CODE REVIEW

File: code/modules/vehicles/pioneer_stone.dm
Comment (nemvar) [inline code/modules/vehicles/pioneer_stone.dm]: This needs typecheck for argument. A can be an atom.

File: code/modules/vehicles/pioneer_stone.dm
Comment (nemvar) [inline code/modules/vehicles/pioneer_stone.dm]: Does this whole section even need to exist?
If so, loop doesn't need to be typed.

Comment (EdgeLordExe) [reply code/modules/vehicles/pioneer_stone.dm]: wouldn't it throw people around without that if(has_buckled_mobs()) whenever someone tried dragging it

Comment (nemvar) [reply code/modules/vehicles/pioneer_stone.dm]: Bump gets called regardless of this blocks existance. At least it should.
If it doesn't, consider not putting this behavior on Bump at all.

Comment (EdgeLordExe) [reply code/modules/vehicles/pioneer_stone.dm]: actually thats true ill try to build it all without the whole block and see if it is needed, as i copied that code from speedwagon

Comment (EdgeLordExe) [reply code/modules/vehicles/pioneer_stone.dm]: it works! damn nemvar thanks a lot

**Comment by Shapsy** [comment]
It's not just a stone, it's a rock!

**Comment by Whoneedspacee** [comment]
i dont really wanna review this but at a glance it looks pretty damn strong for a rock you just find in a random ruin

**Comment by TheVekter** [comment]
PIONEERS USED TO RIDE THESE BABIES FOR MILES

File: code/datums/ruins/lavaland.dm
Comment (ATH1909) [inline code/datums/ruins/lavaland.dm]: "Builder", singular?

File: code/modules/vehicles/pioneer_stone.dm
Comment (ATH1909) [inline code/modules/vehicles/pioneer_stone.dm]: Isn't this going to result in a "sapient non-pAI mulebot with its safeties off"-like situation, where you can just bump into someone in the hall and then repeatedly drive over their prone body until they die, with your victim able to do nothing about it except scream once you get your initial bump in?

Comment (EdgeLordExe) [reply code/modules/vehicles/pioneer_stone.dm]: hm i might just increase the cost of this structure to like 10 or something, so you cannot find it so often. But yeah it is in this situation a little bit

File: code/modules/vehicles/pioneer_stone.dm
Comment (Qustinnus) [inline code/modules/vehicles/pioneer_stone.dm]: this entire codeblock is nested to deep, use early returns so you dont end up 5 tabs indented

File: code/modules/vehicles/pioneer_stone.dm
Comment (optimumtact) [inline code/modules/vehicles/pioneer_stone.dm]: can you check this condition before doing the throw target check work

**Comment by optimumtact** [comment]
how slow is it?

**Comment by EdgeLordExe** [comment]
> how slow is it?

it is approx. the speed of someone is a hardsuit

**Comment by EdgeLordExe** [comment]
> how slow is it?

https://youtu.be/UWlaMmJDNGI here i made a vid for you <3

**Comment by Krysonism** [comment]
I love it.

It will be very strong, especially against cult but i imagine it will be balanced greytiders, clowns and syndicate agents constantly trying to steal your ride.

**Comment by optimumtact** [comment]
needs to take damage or something

**Comment by ishitbyabullet** [comment]
Make it only work on terrain that pioneers would actually ride on.

Who the hell calls themself a pioneer while in the equivalent of an office space or science lab?

**Review by TheChosenEvilOne — CHANGES_REQUESTED** [review]
There already is a proc for checking which z level mining is, also hard coding the mining level z is really bad

File: code/modules/vehicles/pioneer_stone.dm
Comment (TheChosenEvilOne) [inline code/modules/vehicles/pioneer_stone.dm]:
```
var/T = get_turf(src)
if(!is_mining_level(T.z))
```

Comment (EdgeLordExe) [reply code/modules/vehicles/pioneer_stone.dm]: thanks a lot ill update it

File: code/modules/vehicles/pioneer_stone.dm
Comment (Couls) [inline code/modules/vehicles/pioneer_stone.dm]: This means it won’t take damage off the mining z level
Is_mining_level will return 1 when true and 0 when false and anything * 0 is 0

**Review by Couls — CHANGES_REQUESTED** [review]
Reviewing in bed so I can’t fact check but...

File: code/modules/vehicles/pioneer_stone.dm
Comment (EdgeLordExe) [reply code/modules/vehicles/pioneer_stone.dm]: > This means it won’t take damage off the mining z level
> Is_mining_level will return 1 when true and 0 when false and anything * 0 is 0

intended

Comment (Couls) [reply code/modules/vehicles/pioneer_stone.dm]: so you can just run over everyone on station with no damage to the stone and thats intended?

Comment (EdgeLordExe) [reply code/modules/vehicles/pioneer_stone.dm]: wait no , fuck there should be ! in there. my bad

Comment (EdgeLordExe) [reply code/modules/vehicles/pioneer_stone.dm]: you arent supposed to take damage on lavaland
my bad my bad

**Comment by EdgeLordExe** [comment]
god fucking damnit. this pr is finished. ive been waiting for review for a long time. god damnit

File: code/modules/vehicles/pioneer_stone.dm
Comment (Couls) [inline code/modules/vehicles/pioneer_stone.dm]: This probably doesn't need to run every move

**Comment by Couls** [comment]
This thing is kinda op, 5-10 second full potato stun and 60 stamina damage? what stops you from just running someone over constantly or running someone over and then just killing them while they're down for 5 full seconds

**Comment by EdgeLordExe** [comment]
> This thing is kinda op, 5-10 second full potato stun and 60 stamina damage? what stops you from just running someone over constantly or running someone over and then just killing them while they're down for 5 full seconds

it is slower on station and if you hit someone the stone is also damaged

File: code/modules/vehicles/pioneer_stone.dm
Comment (Couls) [inline code/modules/vehicles/pioneer_stone.dm]: don't think you need this because you never called START_PROCESSING anywhere

Comment (Couls) [reply code/modules/vehicles/pioneer_stone.dm]: you can just get rid of the whole destroy proc, the parent code will handle it

Comment (EdgeLordExe) [reply code/modules/vehicles/pioneer_stone.dm]: ill try it.

**Comment by Couls** [comment]
You can still hit 20 people and if you use it in maint its kind of easy to catch people off guard

**Comment by EdgeLordExe** [comment]
> You can still hit 20 people and if you use it in maint its kind of easy to catch people off guard

ill lower the max integrity

**Comment by optimumtact** [comment]
`This may also be an indication that the maintainers do not have interest in this change, you can try to convince them otherwise, or persist in the doomed world you have created.`

**Comment by EdgeLordExe** [comment]
> `This may also be an indication that the maintainers do not have interest in this change, you can try to convince them otherwise, or persist in the doomed world you have created.`

oh i see. thanks for clarification

**Comment by EdgeLordExe** [comment]
Maintainers dont like this fun idea. SAD!

**Pull request closed without merging**
